diff -r -u a/bfd/config.bfd b/bfd/config.bfd
--- a/bfd/config.bfd	2018-07-09 09:44:53.000000000 +0200
+++ b/bfd/config.bfd	2018-11-21 10:42:02.040012456 +0100
@@ -712,6 +712,11 @@
     targ_selvecs=i386_elf32_vec
     want64=true
     ;;
+  x86_64-*-qword*)
+    targ_defvec=x86_64_elf64_vec
+    targ_selvecs=i386_elf32_vec
+    want64=true
+    ;;
 #endif
   i[3-7]86-*-lynxos*)
     targ_defvec=i386_elf32_vec
diff -r -u a/config.sub b/config.sub
--- a/config.sub	2018-07-06 09:13:31.000000000 +0200
+++ b/config.sub	2018-11-21 10:39:46.638332604 +0100
@@ -1376,7 +1376,8 @@
 	     | powermax* | dnix* | nx6 | nx7 | sei* | dragonfly* \
 	     | skyos* | haiku* | rdos* | toppers* | drops* | es* \
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
-	     | midnightbsd*)
+	     | midnightbsd* \
+	     | qword*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	qnx*)
diff -r -u a/gas/configure.tgt b/gas/configure.tgt
--- a/gas/configure.tgt	2018-06-24 20:38:57.000000000 +0200
+++ b/gas/configure.tgt	2018-11-21 10:45:26.376525655 +0100
@@ -6,12 +6,12 @@
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation; either version 3 of the License, or
 # (at your option) any later version.
-# 
+#
 # This program is distributed in the hope that it will be useful,
 # but WITHOUT ANY WARRANTY; without even the implied warranty of
 # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 # GNU General Public License for more details.
-# 
+#
 # You should have received a copy of the GNU General Public License
 # along with this program; see the file COPYING3.  If not see
 # <http://www.gnu.org/licenses/>.
@@ -255,6 +255,7 @@
   i386-*-chaos)				fmt=elf ;;
   i386-*-rdos*)				fmt=elf ;;
   i386-*-darwin*)			fmt=macho ;;
+  i386-*-qword*)			fmt=elf ;;
 
   ia16-*-elf*)				fmt=elf ;;
 
@@ -322,7 +323,7 @@
   moxie-*-uclinux)			fmt=elf em=linux ;;
   moxie-*-moxiebox*)                    fmt=elf endian=little ;;
   moxie-*-*)				fmt=elf ;;
-    
+
   mt-*-elf)				fmt=elf bfd_gas=yes ;;
 
   msp430-*-*)				fmt=elf ;;
@@ -408,7 +409,7 @@
   wasm32-*-*)                           fmt=elf ;;
 
   xstormy16-*-*)			fmt=elf ;;
-  
+
   xgate-*-*)    			fmt=elf ;;
 
   xtensa*-*-*)				fmt=elf ;;
diff -r -u a/ld/configure.tgt b/ld/configure.tgt
--- a/ld/configure.tgt	2018-07-09 09:44:53.000000000 +0200
+++ b/ld/configure.tgt	2018-11-21 10:51:16.922514170 +0100
@@ -260,6 +260,10 @@
 			targ_extra_emuls=elf_x86_64 ;;
 x86_64-*-redox*)	targ_emul=elf_x86_64
 			targ_extra_emuls=elf_i386 ;;
+x86_64-*-qword*)
+			targ_emul=elf_x86_64
+			targ_extra_emuls=elf_i386
+			;;
 i[3-7]86-*-sysv[45]*)	targ_emul=elf_i386
 			targ_extra_emuls=elf_iamcu ;;
 i[3-7]86-*-solaris2*)	targ_emul=elf_i386_sol2
