diff -urN ncurses-6.1/config.sub ncurses-6.1-patched/config.sub
--- ncurses-6.1/config.sub	2017-11-25 21:49:31.000000000 +0100
+++ ncurses-6.1-patched/config.sub	2019-05-06 21:13:44.158279190 +0200
@@ -1416,7 +1416,7 @@
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
-	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox*)
+	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* | -qword*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
diff -urN ncurses-6.1/ncurses/tty/lib_twait.c ncurses-6.1-patched/ncurses/tty/lib_twait.c
--- ncurses-6.1/ncurses/tty/lib_twait.c	2016-05-29 01:32:40.000000000 +0200
+++ ncurses-6.1-patched/ncurses/tty/lib_twait.c	2019-05-06 21:14:29.652279923 +0200
@@ -75,6 +75,8 @@
 #endif
 #undef CUR
 
+#include <sys/time.h>
+
 MODULE_ID("$Id: lib_twait.c,v 1.71 2016/05/28 23:32:40 tom Exp $")
 
 static long
