diff -urN a/build-aux/config.sub b/build-aux/config.sub
--- a/build-aux/config.sub	2018-06-24 06:52:06.000000000 +0200
+++ b/build-aux/config.sub	2019-01-07 06:35:49.847947740 +0100
@@ -1376,7 +1376,8 @@
 	     | powermax* | dnix* | nx6 | nx7 | sei* | dragonfly* \
 	     | skyos* | haiku* | rdos* | toppers* | drops* | es* \
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
-	     | midnightbsd*)
+	     | midnightbsd* \
+	     | qword* )
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	qnx*)
diff -urN a/configure b/configure
--- a/configure	2018-07-02 02:56:53.000000000 +0200
+++ b/configure	2019-01-07 07:00:31.964633065 +0100
@@ -43399,7 +43399,7 @@
 fi
 
 if test -z "$ac_list_mounted_fs"; then
-  as_fn_error $? "could not determine how to read list of mounted file systems" "$LINENO" 5
+  echo 'as_fn_error $? "could not determine how to read list of mounted file systems" "$LINENO" 5'
   # FIXME -- no need to abort building the whole package
   # Can't build mountlist.c or anything that needs its functions
 fi
